spring-integration-redis/src/main/java/org/springframework/integration/redis/util/RedisLockRegistry.java

```diff
@@ -301,6 +305,10 @@ public void unlock() {
 				return;
 			}
 			try {
+				if (!isAcquiredInThisProcess()) {
+					throw new IllegalStateException("Lock was released in the store due to expiration.");
```

Comment:
Perhaps we should be more explicit? `...due to expiration. The integrity of data protected by this lock may have been compromised.` ?